recommend/policyTemplates.go

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/cavaliergopher/grab/v3"
+	"github.com/google/go-github/github"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	org   = "kubearmor"
+	repo  = "policy-templates"
+	url   = "https://github.com/kubearmor/policy-templates/archive/refs/tags/"
+	cache = ".cache/karmor/"
+)
+
+var currentVersion string
+
+func getCachePath() string {
+	return cache
+
+}
+
+// userHome function returns users home directory
+func userHome() string {
+	if runtime.GOOS == "windows" {
+		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
+		if home == "" {
+			home = os.Getenv("USERPROFILE")
+		}
+		return home
+	}
+	return os.Getenv("HOME")
+}
+
+// isTemplates function returns error if policy-template folder not found
+func isTemplates() error {
+	path := fmt.Sprintf("%s/%s", userHome(), getCachePath())
+	if stat, err := os.Stat(path); err == nil && stat.IsDir() {
+		if !isLatest() {
+			log.Warn("\033[1;33mA new version of policy-templates is available. Use `karmor recommend --update` to get recommendations based on the latest policy-templates.\033[0m")
+		}
+	} else {
+		log.Warn("\033[1;33mNo version of policy-templates available locally. Use `karmor recommend --update` to get recommendations based on the latest policy-templates.\033[0m")
+	}
+	return nil
+}
+
+func latestRelease() string {
+	latestRelease, _, err := github.NewClient(nil).Repositories.GetLatestRelease(context.Background(), org, repo)
+	if err != nil {
+		log.WithError(err)
+		return ""
+	}
+	return *latestRelease.TagName
+}
+func currentRelease() string {
+	path := fmt.Sprintf("%s/%s", userHome(), getCachePath())
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return "nil"
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			currentVersion = fmt.Sprintf("v%s", strings.Split(file.Name(), "policy-templates-")[1])
```

Comment:
We should keep the store/retrieve the current version from `rules.yaml`.
```yaml
version: 0.0.1
policySpec:
- <MatchSpec-1>
- <MatchSpec-2>
```
In future, the generation of `rules.yaml` that ships with `karmor` binary by default, will be automated. The CI will automatically take the latest policy-template release, generate `rules.yaml`, set the `version` field to latest release version of policy-template and embedded that data into `karmor`. 

**(To-Do)**
If we want to incorporate this idea, then when we generate `rules.yaml` while `karmor recommend update`, we should add the version as part of `rules.yaml`. And when we call `updateRulesYAML` we should the parse the `version` and store it somewhere.